### DISCUSSION TITLE
Adding / removing tracks from circular view

### DISCUSSION THREAD

**Discussion by jrobinso** [GMOD/jbrowse-components#2477] [General]
Can I dynamically add (and remove) new tracks to the circular view?  I know how to add or replace features in an existing track, but since tracks are listed in the view that comes up from the "hamburger" menu users are going to expect there features segregated this way.  

I naively tried this, but get an error that tracks cannot be modified except through an "action"

```
    addTrack(name, chords) {
        const track = {
            trackId: name,
            name: name,
            assemblyNames: ['forIGV'],
            type: 'VariantTrack',
            adapter: {
                type: 'FromConfigAdapter',
                features: chords,
            }
        }
        this.viewState.config.tracks.push(track);  // <<< DOESN't WORK
    }
```

Again I can point you to a full example if the question isn't clear.   Thanks.

**Comment by cmdcolin** [comment]
in general, the mobx-state-tree machinery needs to use "actions" to modify the state so we would want to create a function on the viewState model called e.g. viewState.addTrack to do the same action you are doing here

we hadn't implemented track adding to viewState yet but will look into it
